hybrasyl/Messaging/Admin.cs

```diff
@@ -0,0 +1,410 @@
+                    var target = Game.World.WorldData.Get<User>(args[1]);
+                    user.Teleport(target.MapId, target.X, target.Y);
+                    return Success($"Teleported to {target.Name} - {target.Map.Name} ({target.X},{target.Y})");
+                }
+            }
+            else
+            {
+                if (ushort.TryParse(args[0], out ushort mapnum) && !byte.TryParse(args[1], out byte x) && !byte.TryParse(args[2], out byte y))
+                {
+
+                    if (Game.World.WorldData.ContainsKey<Map>(mapnum))
+                    {
+                        var map = Game.World.WorldData.Get<Map>(mapnum);
+                        if (x < map.X && y < map.Y)
+                        {
+                            user.Teleport(mapnum, x, y);
+                            return Success($"Teleported to {map.Name} ({x},{y}).");
+                        }
+                        else
+                            return Fail("Invalid x/y specified (hint: mapsize is {map.X}x{map.Y})");
+                    }
+                    else
+                        return Fail("Map number {mapnum} not found");
+                }
+                else
+                {
+                    return Fail("Couldn't parse map number or coordinates (uint / byte / byte)");
+                }
+            }
+        }
+    }
+
+    class DebugCommand : ChatCommand
+    {
+        public new static string Command = "debug";
+        public new static string ArgumentText = "None";
+        public new static string HelpText = "Toggle whether or not debugging is enabled on the server.";
+        public new static bool Privileged = true;
+
+        public new static ChatCommandResult Run(User user, params string[] args)
+        {
+            var enabled = Game.World.ToggleDebug();
+            if (enabled)
+                return Success("Debugging enabled");
+            return Success("Debugging disabled");
+        }
+    }
+
+    class SpawnCommand : ChatCommand
+    {
+        public new static string Command = "spawn";
+        public new static string ArgumentText = "<string monsterName> [<uint hp> <uint mp> <uint str> <uint int> <uint wis> <uint con> <uint dex> <uint xp> <uint gold>]";
+        public new static string HelpText = "Spawn the specified monster at your current coordinates, with optionally specified stats.";
+        public new static bool Privileged = true;
+
+        public new static ChatCommandResult Run(User user, params string[] args)
+        {
+
+            if (Game.World.WorldData.TryGetValue(args[0], out Creatures.Creature creature))
+            {
+                Spawn spawn = new Spawn();
+                spawn.Stats.Hp = 100;
+                spawn.Stats.Mp = 100;
+                spawn.Stats.Str = 3;
+                spawn.Stats.Int = 3;
+                spawn.Stats.Wis = 3;
+                spawn.Stats.Con = 3;
+                spawn.Stats.Dex = 3;
+                spawn.Loot.Xp.Min = 1;
```

Comment:
Xp and Gold assignments are getting NullExceptions. Looks like it's trying to get the value, but not sure why, instead of setting it. Also, spawn needs to have a List<> assigned to it, otherwise, it's also generating a NullException when it gets to Monster->Line162 [CanCast = spawn.Castables.Count > 0].

Reply:
fixed